lib/AST/Type.cpp

```diff
@@ -2027,9 +2027,15 @@ static ForeignRepresentableKind
 getObjCObjectRepresentable(Type type, const DeclContext *dc) {
   // @objc metatypes are representable when their instance type is.
   if (auto metatype = type->getAs<AnyMetatypeType>()) {
+    auto instanceType = metatype->getInstanceType();
+
+    // Consider protocol metatypes as not Objective-C representable.
+    if (isa<MetatypeType>(metatype) &&
+        instanceType->getKind() == TypeKind::Protocol)
```

Comment:
This is good but it won't catch a typealias to a protocol, whose TypeKind will be something else. The best mechanism for dealing with that is the`TypeBase::getAs` function, which will look for a "canonical" type instead of just checking the outermost type.

Can you add a test for typealiases as well?

```swift
typealias AnotherNSCoding = NSCoding
typealias MetaNSCoding = NSCoding.Protocol
```